specs/anchored-region/anchored-region.md

```diff
@@ -0,0 +1,254 @@
+    >
+      This shows up above or below the anchor depending on available space
+    </anchored-region>
+    ...stuff...
+</div>
+
+```
+
+A region that renders above or below the anchor depending on available space but only on initial layout.
+```
+<div id="viewport">
+    ...stuff...
+    <button id="anchor">
+        Button is an anchor
+    </button>
+    <anchored-region
+        anchorid="anchor"
+        verticalpositioningmode="onetime"
+    >
+      This shows up above or below the anchor depending on available space
+    </anchored-region>
+    ...stuff...
+</div>
+
+```
+
+A region that overlaps the anchor and renders above or below it depending on available space.
+
+```
+<div id="viewport">
+    ...stuff...
+    <button id="anchor">
+        Button is an anchor
+    </button>
+    <anchored-region
+        anchorid="anchor"
+        verticalPositioningMode="dynamic"
+        verticalinset="true"
+    >
+      This overlaps the anchor and extends above or below it depending on available space.
+    </anchored-region>
+    ...stuff...
+</div>
+
+```
+
+A region renders above or below the anchor based on available space and is sized to match the available.
+
+```
+<div id="viewport">
+    ...stuff...
+    <button id="anchor">
+        Button is an anchor
+    </button>
+    <anchored-region
+        anchorid="anchor"
+        verticalPositioningMode="dynamic"
+        verticalscalingenabled="true"
+    >
+       This region renders above or below the anchor based on available space and sizes itself to match.
+    </anchored-region>
+    ...stuff...
+</div>
+
+```
+
+A region that renders below the anchor until that space is less than 100px.
+```
+<div id="viewport">
+    ...stuff...
+    <button id="anchor">
+        Button is an anchor
+    </button>
+    <anchored-region
+        anchorid="anchor"
+        verticalpositioningmode="dynamic"
+        verticaldefaultposition="below"
+        verticalthreshold="100"
+    >
+      This shows renders below the anchor as long as there is at least 100px available there.
+    </anchored-region>
+    ...stuff...
+</div>
+```
+
+The dimensions of the anchored region will match the dimensions of the content unless scaling is enabled on a particular axis (verticalscalingenabled & horizontalscalingenabled) in which case it will fill all available space between the anchor and viewport.
+
+### API
+
+*Component name:*
+- `fast-anchored-region`
+
+*Attributes:*
+- anchorId - The html id of the HTMLElement used as the anchor around which the positioning region is placed.  This must be set for the component's positioning logic to be active.
+
+- viewportId - The html id of the HTMLElement used as the viewport the component uses to determine available layout space around the anchor element.  If unset the immediate parent element of the positioner is used.
+
+- horizontalPositioningMode - Can be 'uncontrolled', 'locktodefault', 'dynamic' or 'onetime'.  Default is 'uncontrolled'.
```

Comment:
Do you describe what the values 'uncontrolled', 'locktodefault', 'dynamic' and 'onetime' do? It's not super clear to me what effect these will have.

Reply:
Adding this to the "Design" section:

The component allows users to set a "Positioning Mode" on each axis which defines how the component will behave:
- 'uncontrolled':  The anchored region will appear as it normally would in document flow.
- 'locktodefault': The anchored region will always be placed in the specified default position regardless of available space.  For example a menu that always opens upwards.
- 'dynamic': The anchored region is placed relative to the anchor element based on how much space is available between it and the viewport.  For example a menu that opens opens upwards when it is near the bottom of the page, and downwards when near the top.
- 'onetime': As the 'dynamic' option, but position is not changed after initial placement.

Reply:
Can you add them as a list under this bullet point so we can directly refer to them?

Reply:
There are two "positioning mode" attributes so I'd rather not duplicate the detailed description.  It immediately precedes the api section of the doc so it is ver close